test/integration/instance.validations.test.ts

```diff
@@ -0,0 +1,978 @@
+'use strict';
+
+import type {
+  InferAttributes,
+  CreationOptional,
+  InferCreationAttributes,
+  HasManyAddAssociationMixin,
+  BelongsToSetAssociationMixin,
+  HasMany,
+  BelongsTo,
+  Utils,
+} from '@sequelize/core';
+import {
+  DataTypes,
+  ValidationError,
+  Model,
+  Sequelize,
+} from '@sequelize/core';
+import chai from 'chai';
+import {
+  getTestDialectTeaser,
+  getTestDialect,
+  sequelize,
+  beforeEach2,
+} from './support';
+
+const expect = chai.expect;
+
+describe(getTestDialectTeaser('InstanceValidator'), async () => {
+  describe('#update', async () => {
+    it('should allow us to update specific columns without tripping the validations', async () => {
+      class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+        declare id?: CreationOptional<number>;
+        declare username: string;
+        declare email: string;
+      }
+
+      User.init({
+        username: DataTypes.STRING,
+        email: {
+          type: DataTypes.STRING,
+          allowNull: false,
+          validate: {
+            isEmail: {
+              msg: 'You must enter a valid email address',
+            },
+          },
+        },
+      }, { sequelize });
+
+      await User.sync({ force: true });
+
+      const user = await User.create({ username: 'bob', email: 'hello@world.com' });
+
+      await User
+        .update({ username: 'toni' }, { where: { id: user.id } });
+
+      const user0 = await User.findByPk(1);
+      expect(user0?.username).to.equal('toni');
+    });
+
+    it('should be able to emit an error upon updating when a validation has failed from an instance', async () => {
+      class Book extends Model<InferAttributes<Book>, InferCreationAttributes<Book>> {
+        declare name: string;
+      }
+
+      Book.init({
+        name: {
+          type: DataTypes.STRING,
+          allowNull: false,
+          validate: {
+            notEmpty: true, // don't allow empty strings
+          },
+        },
+      }, { sequelize });
+
+      await Book.sync({ force: true });
+      const book = await Book.create({ name: 'World' });
+
+      try {
+        await book.update({ name: '' });
+      } catch (error) {
+        expect(error).to.be.an.instanceOf(Error);
+        if (error instanceof ValidationError) {
```

Comment:
If it's supposed to be a ValidationError, line 83 should check that it's an instanceof ValidationError, not Error, and should use `assert` instead of expect for proper type narrowing

Same comment for the following tests